draft-ietf-taps-impl.md

```diff
@@ -1030,39 +1048,78 @@ Data Unit: Message
-To map Connection objects to SCTP streams without head-of-line blocking on the sender
-side, both the sending and receiving SCTP implementation must support message interleaving {{!RFC8260}}.
-Both SCTP implementations must also support stream reconfiguration. Finally, both communicating endpoints
-must be aware of this intended multiplexing; {{NEAT-flow-mapping}} describes a
-way for a Transport System to negotiate the stream mapping capability using SCTP's adaptation layer indication,
-such that this functionality would only take effect if both ends sides are aware of it.
-The first flow, for which the SCTP association has been created, will always use stream id zero.
-All additional flows are assigned to unused stream ids in growing order. To avoid a conflict
-when both endpoints map new flows simultaneously, the peer which initiated the transport connection
-will use even stream numbers whereas the remote side will map its flows to odd stream numbers.
-Both sides maintain a status map of the assigned stream numbers. Generally, new streams
-must consume the lowest available (even or odd, depending on the side) stream number; this
-rule is relevant when lower numbers become available because Connection objects associated
-to the streams are closed.
+: Connection objects can be an SCTP association or a stream in an SCTP association.
+We call the process of mapping Connection objects to SCTP streams "stream mapping",
+and assume a client-server communication model in the following explanation.
+
+Stream mapping requires an association to already be in place between the client
+and the server, and it requires the server to understand that a new incoming stream
+should be represented as a new Connection Object by the Transport Services system.
+To signal the use of stream mapping, the server SHOULD use the adaptation layer indication
+parameter of LISTEN.SCTP and the client SHOULD use the adaptation layer indication
+parameter of CONNECT.SCTP, both with the "Stream Mapping" adaptation code point.
+Reception of an adaptation layer indication notification with same adaptation code point value
+then indicates that the peer is aware of stream mapping.
+
+In the absence of an adaptation layer indication notification carrying a "Stream Mapping"
+daptation code point, stream mapping can still be
+used if the application is aware that the peer application does not directly access SCTP streams.
+If, for example, a transport-protocol-agnostic client application tries to connect to a
+native SCTP server application that does not implement stream mapping, and assumes
+that a new Message sent on this Connection will look like the first Message of a new
+Connection on the server side, communication will fail because the
+SCTP-based application on the server side does not obtain a new Connection.
+As a counter-example, if both the client and server are
+written over a Transport Services system, and the server listens on all available protocols, stream
```

Comment:
When did this section become about disambiguating different protocols? I thought it was just about different associations/streams?

Reply:
I don't understand this failure mode. If for whatever reason the peer doesn't accept the stream, how is this different from the peer rejecting due to HTTP/2 or QUIC stream limits?

Reply:
No point discussing - it's related to the codepoint stuff that goes out.